src/contents/shared/ProfileEdit.js

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { Component } from 'react';
+import fire from '../../config/Fire';
+import { ProfileEditView } from './ProfileEditView';
+import firebase from 'firebase';
+
+export class ProfileEdit extends Component {
+	state = {
+		  image: "https://example.com/jane-q-user/profile.jpg",
+		  url: this.props.user.photoURL,
+		  progress: 0,
+		  name: this.props.user.displayName,
+		  email: this.props.user.email,
+		  type: this.props.user.type,
+		  location: this.props.user.location,
+		  photoURL: this.props.user.url,
+		}
+	  
+	database = fire.database().ref();
+
+	/**
+		* Updates state to the current value of a certain target.
+		* @param e
+		*/
+		handleChange = e => {
+			this.setState({ [e.target.name]: e.target.value });
+		};
+
+	/**
+		* Registers the user as photographer or company.
+		*
+		* @param e
+		*/
+
+	updateUser = (e) => {
+		e.preventDefault();
+		let { name, email, type, location, url } = this.state;
+		const { user } = this.props;
+
+		if (name !== "" && email !== "" && location !== "" && url !== "") {
+
+		firebase.auth().currentUser.updateProfile(
+			{
+
+				displayName: name,
+				photoURL: url
+			}
+		)
+
+			.then(() => {
+
+				if(user.location !== location && location !=="") {
+					this.database.child("locations").child(user.location).child(user.type).child(user.uid).remove();
+					
+					
+					this.database.child("locations").child(location).child(user.type).child(user.uid).update({
+						displayName: name,
+						photoURL: url
+					});
+				}
+				else {
+					console.log("location was same as before so no update:" + user.location);
+
+				}
+
+			})
+
+			.then(() => {
+				this.database.child("users").child(user.uid).update({
+					type: type,
+					email: email,
+					location: location,
+					displayName: name,
+					photoURL: url
+				});
+
+			})
+
+			.then(() => {
+				this.database.child(user.type).child(user.uid).update({
+					email: email,
+					location: location,
+				});
+			})
+
+			.then(() => {
+				this.props.history.replace("/dashboard");
+				alert("User Updated!! please refresh the page for updates");
+			  })
+	
+		}
+		else {
+			alert("please fill in all the input fields");
+		}
+	};
+
+	handleChangeUpload = e => {
+		this.setState({ [e.target.name]: e.target.value });
+		if (e.target.files[0]) {
+		  const image = e.target.files[0];
+		  this.setState(() => ({image}));
+
+		  //handle start
+		  
+		
+		const uploadTask = fire.storage().ref(`images/${image.name}`).put(image);
```

Comment:
We save the files like so , for each user in the storage, we create a folder with his id, then inside there we have a folder for the portfolio and another file called avatar which is his avatar, it will be much easier to mentain , otherwise if you save the images by their name in a general folder, then it might be that 2 users have same name on their files which will result into a bug.
So do it like so ref(${userId}/avatar) then we can easily update / delete them

Reply:
Done!